Add jcstress test to system test

Mesbah-Alam opened the pull request:
Related : https://github.com/adoptium/aqa-tests/issues/3546

Should be delivered after: https://github.com/adoptium/TKG/pull/321

Signed-off-by: Mesbah-Alam <Mesbah_Alam@ca.ibm.com>

Mesbah-Alam commented:
Test : Grinder/23850

llxia commented on system/jcstress/playlist.xml:
```diff
@@ -0,0 +1,31 @@
+		<command>$(JAVA_COMMAND) $(JVM_OPTIONS) -jar $(Q)${TEST_ROOT}$(D)systemtest_prereqs$(D)jcstress$(D)jcstress-latest.jar$(Q) -jvmArgs $(Q)$(JVM_OPTIONS) -t SampleTestBench$(Q); \
+		$(TEST_STATUS)</command>
+		<versions>
+			<version>11</version>
```
Should this be 8+? If so, we do not need the version tag.

Mesbah-Alam replied:
Good catch. Updated, works. Grinder/23855.

llxia commented:
Ideally, we should separate jcstress with STF as they are unrelated. However, this PR creates the dependency between jcstress and STF (i.e, we cannot run jcstress without STF). And without build.xml, I suspect dynamic compilation will not work in this case.

I am ok with the PR for now as this enables us to run jcstress. Please open an issue for decoupling jcstress and STF. I think the better way is to use getDependencies in TKG.

Mesbah-Alam commented:
> we cannot run jcstress without STF

We can run jcstress without STF. The only dependency is the top.xml changes (https://github.com/adoptium/STF/pull/127). However, since we want to run jckstress as part of systemtest, and system test will compile STF anyway, there is no extra cost of having the dependency on top.xml to fetch jcstress.jar.

smlambert commented:
Ideally for the dynamic compilation feature that TKG has, if I only wanted to run the jcstress target by itself, then it should not have to go and compile STF but agree any further decoupling can be in a 'phase 2'.

Mesbah-Alam commented:
Sounds good! Instead of doing it in phase 2 , let's move to using the getDependencies script in TKG now. 

I've added jcstress jar to the list of downloads in get Dependencies script in TKG : https://github.com/adoptium/TKG/pull/321. 

Manually ran test.getDependency job pointing to my branch for testing : https://hyc-runtimes-jenkins.swg-devops.com/job/test.getDependency/10261/. 

(Note, once https://github.com/adoptium/TKG/pull/321 is delivered, we need to run the [test.getDependency job at ci.adopt](https://ci.adoptopenjdk.net/job/test.getDependency/)). 

Updated the PR for playlist. New test : Grinder/23939.

llxia commented:
Thanks @Mesbah-Alam . 

Please also add build.xml file in this case. Otherwise, I think the local run will fail. :) 
The build.xml should contain the following. It will trigger getDependencies for downloading jcstress if needed.
```
	<property name="LIB" value="jcstress" />
	<import file="${TEST_ROOT}/TKG/scripts/getDependencies.xml" />
```

Mesbah-Alam commented:
Re: https://github.com/adoptium/aqa-tests/pull/3666#issuecomment-1130404099

Added. Please check if this will suffice!

llxia commented on system/jcstress/build.xml:
```diff
@@ -0,0 +1,20 @@
+<project name="JCStressTest" basedir=".">
+	<property name="LIB" value="jcstress" />
+	<import file="${TEST_ROOT}/TKG/scripts/getDependencies.xml" />
+</project>
```
Sorry, the above is not sufficient. We need the default targets.

```
<project name="JCStressTest" default="build" basedir=".">
	<taskdef resource="net/sf/antcontrib/antlib.xml" />
	<description>JCStress Test</description>

	<!-- Set global properties for this build -->
	<property name="DEST" value="${BUILD_ROOT}/system/jcstress" />
	<property name="SRC" location="." />

	<property name="LIB" value="jcstress" />
	<import file="${TEST_ROOT}/TKG/scripts/getDependencies.xml" />

	<target name="init">
		<mkdir dir="${DEST}" />
	</target>
	
	<target name="dist" depends="init" description="generate the distribution">
		<copy todir="${DEST}">
			<fileset dir="${SRC}"/>
		</copy>
	</target>

	<target name="build">
		<antcall target="dist" inheritall="true" />
	</target>
</project>
```

llxia replied:
This should be tested via `BUILD_LIST=system/jcstress`

Mesbah-Alam replied:
Added (tested: Grinder/23946). Please check.

Mesbah-Alam replied:
Here's the test with `BUILD_LIST=system/jcstress` : Grinder/24008.

llxia reviewed (approved):
LGTM

smlambert reviewed (approved):
Since the related TKG PR is merged, I will proceed to merge this one.  Thanks @Mesbah-Alam for this initial add, and as discussed, we can now refine what we run out of the jcstress repo quite easily in a follow-up PR. 👍
